components/automatic-points-table/commons/automatic-points-table.lua

```diff
@@ -54,4 +58,76 @@ function AutomaticPointsTable:parseTournaments(args)
 	return tournaments
 end
 
+function AutomaticPointsTable:parseTeams(args)
+	local teams = {}
+	for _, team in Table.iter.pairsByPrefix(args, 'team') do
+		local parsedTeam = Json.parse(team)
+		parsedTeam.aliases = self:parseAliases(parsedTeam)
+		parsedTeam.deductions = self:parseDeductions(parsedTeam)
+		parsedTeam.manualPoints = self:parseManualPoints(parsedTeam)
+		table.insert(teams, parsedTeam)
+	end
+	return teams
+end
+
+--- Parses the team aliases, used in cases where a team is picked up by an org or changed name in some
+--- of the tournaments, in which case aliases are required to correctly query the team's results & points
+function AutomaticPointsTable:parseAliases(team)
+	local aliases = {}
+	for key, value in pairs(team) do
+		if type(key) == 'string' and string.find(key, 'alias%d+') then
+			local aliasIndexString = String.split(key, 'alias')[1]
+			local aliasIndex = tonumber(aliasIndexString)

```

Comment:
Use pairsByPrefix

Reply:
It's not guaranteed that if there's **alias2** there will be **alias1** which means the function from Table module wouldn't work

Reply:
Ah wait, is aliasX is alias for team X?

Reply:
Nope, aliasX is the alias for some team in the tournament whose index is X
Here's an example
```
  |team3={{#invoke:Json|fromArgs|name=Ex-Barça
    |alias4=FC Barcelona
    |deduction1=10
    |deduction4=720
    |deduction4note=Ex-Barça lose 720 points after the Fall Major due to a Team disband
  }}
```

Reply:
Yeah I realized after writing the first time (and edited above) (incase anyone reading the conversation later is and confused)

Reply:
> Nope, aliasX is the alias for some team in the tournament which index is X Here's an example

In that case, I think it would be better then to iterate 
```
for index = 1, #tournaments do
	if String.isNotEmpty(team['alias'.. index]) then
		aliases[index] = team['alias'.. index]
	end
end
```

Reply:
This popped in my mind as well, seems like a better solution so I'll implement